exchanges/binanceus/binanceus_wrapper.go

```diff
@@ -0,0 +1,1010 @@
+				Close:        tick[y].PrevClosePrice,
+				Pair:         pairFmt,
+				ExchangeName: bi.Name,
+				AssetType:    a,
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// FetchTicker returns the ticker for a currency pair
+func (bi *Binanceus) FetchTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	fPairs, er := bi.FormatExchangeCurrency(p, assetType)
+	if er != nil {
+		return nil, er
+	}
+	bi.appendOptionalDelimiter(&fPairs)
+
+	tickerNew, er := ticker.GetTicker(bi.Name, p, assetType)
+	if er != nil {
+		return bi.UpdateTicker(ctx, p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// appendOptionalDelimiter ensures that a delimiter is
+// present for long character currencies
+func (bi *Binanceus) appendOptionalDelimiter(p *currency.Pair) {
+	if len(p.Quote.String()) > 3 ||
+		len(p.Base.String()) > 3 {
+		p.Delimiter = ":"
+	}
+}
+
+// FetchOrderbook returns orderbook base on the currency pair
+func (bi *Binanceus) FetchOrderbook(ctx context.Context, pair currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	fPair, err := bi.FormatExchangeCurrency(pair, assetType)
+	if err != nil {
+		return nil, err
+	}
+	bi.appendOptionalDelimiter(&fPair)
+	ob, err := orderbook.Get(bi.Name, pair, assetType)
+	if err != nil {
+		return bi.UpdateOrderbook(ctx, pair, assetType)
+	}
+	return ob, nil
+}
+
+// UpdateOrderbook updates and returns the orderbook for a currency pair
+func (bi *Binanceus) UpdateOrderbook(ctx context.Context, pair currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	book := &orderbook.Base{
+		Exchange:        bi.Name,
+		Pair:            pair,
+		Asset:           assetType,
+		VerifyOrderbook: bi.CanVerifyOrderbook,
+	}
+
+	orderbookNew, err := bi.GetOrderBookDepth(ctx, &OrderBookDataRequestParams{
+		Symbol: pair,
+		Limit:  1000})
+	if err != nil {
+		return book, err
+	}
+
+	book.Bids = make([]orderbook.Item, len(orderbookNew.Bids))
+	for x := range orderbookNew.Bids {
+		book.Bids[x] = orderbook.Item{
+			Amount: orderbookNew.Bids[x].Quantity,
+			Price:  orderbookNew.Bids[x].Price,
+		}
+	}
+
+	book.Asks = make([]orderbook.Item, len(orderbookNew.Asks))
+	for x := range orderbookNew.Asks {
+		book.Asks[x] = orderbook.Item{
+			Amount: orderbookNew.Asks[x].Quantity,
+			Price:  orderbookNew.Asks[x].Price,
+		}
+	}
+	err = book.Process()
+	if err != nil {
+		return book, err
+	}
+	return orderbook.Get(bi.Name, pair, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (bi *Binanceus) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
+	var info account.Holdings
+	var acc account.SubAccount
+	info.Exchange = bi.Name
+	if assetType != asset.Spot {
+		return info, fmt.Errorf("%v  assetType is not supported", assetType)
+	}
+	theAccount, err := bi.GetAccount(ctx)
+	if err != nil {
+		return info, err
+	}
+	var currencyBalance []account.Balance
```

Comment:
This linter wants this to be `currencyBalance := make([]account.Balance, len(theAccount.Balances))`

I know I just told you to change this :smile: